en/02_Developer_Guides/02_Controllers/02_Routing.md

```diff
@@ -303,6 +303,97 @@ class BreadAPIController extends Controller
+
+`RequestHandler` is the base class for other classes that can handle HTTP requests such as `Controller`, [`FormRequestHandler`](api:SilverStripe\Forms\FormRequestHandler) (used by [`Form`](api:SilverStripe\Forms\Form)) and [`FormField`](api:SilverStripe\Forms\FormField).
+
+### How it works
+
+[`Director::handleRequest()`](api:SilverStripe\Control\Director::handleRequest()) begins the URL parsing process by parsing the start of the URL and workng out which request handler subclass to use by looking in routes set in YAML config under `Director.rules`.
+
+When a request handler matching the first portion of the URL is found, the `handleRequest()` method on the matched request handler subclass is called. This passes control to the matched request handler and the URL is effectively truncated from the left, removing the portion that lead to ths point.
+
+From there regular request handling occurs and the URL will be checked to see if it matches `$allowed_actions` on the `RequestHandler`, possibly routed via `$url_handlers`. If an `$allowed_action` (i.e. method on the `RequestHandler`) is matched and that method returns a request handler, then control will be passed to this nested request handler and the URL is again effectively truncated from the left.
+
+### Example of anested request handler being returned in an action method
+
+Using the code below, navigating to the URL `/one/two/hello` will return a response with a body of "hello"
+
+```yml
+// app/_config/routes.yml
+
+SilverStripe\Control\Director:
+  rules:
+    'one': 'RequestHandlerOne'
+```
+
+```php
+// app/src/RequestHandlerOne.php
+
+use SilverStripe\Control\Controller;
+use SilverStripe\Control\HTTPRequest;
+
+class RequestHandlerOne extends Controller
+{
+    private static $allowed_actions = [
+        'two',
+    ];
+
+    public function two(HTTPRequest $request)
+    {
+        return RequestHandlerTwo::create();
+    }
+}
+```
+
+```php
+// app/src/RequestHandlerTwo.php
+
+use SilverStripe\Control\Controller;
+use SilverStripe\Control\HTTPRequest;
+use SilverStripe\Control\HTTPResponse;
+
+class RequestHandlerTwo extends Controller
+{
+    private static $allowed_actions = [
+        'hello',
+    ];
+
+    public function hello(HTTPRequest $request)
+    {
+        return HTTPResponse::create()->setBody('hello');
+    }
+}
+```
+
+### How `RequestHandler` and `Form` work together
+
+`Form` does not extend `RequestHandler`, instead it implements the [`HasRequestHandler`](api:SilverStripe\Control\HasRequestHandler) interface which defines a method `getRequestHandler()`. [`Form::getRequestHandler()`](api:SilverStripe\Forms\Form::getRequestHandler()) returns a `FormRequestHandler` which is a subclass of `RequestHandler`.
+
+Request handler subclasses and implementors of `HasRequestHandler` are treated the same because they will both end up calling `handleRequest()` on the subclass of the appropriate request handler subclass.
```

Comment:
```suggestion
Request handlers and implementors of `HasRequestHandler` are treated the same because they will both end up calling `handleRequest()` on the subclass of the appropriate request handler.
```

Reply:
Updated